### ISSUE TITLE
Alerting: Unified Alerting REST API: Field `For` is always set to `0`

### ISSUE THREAD

**Issue by dfoxg** [grafana/grafana#53183]
**What happened**:
I´m trying to create a Grafana-Managed rule via the unified alerting API. The value for the field `for` seems to be ignored. My JSON i send to `/api/v1/provisioning/alert-rules`:
```
 "annotations": {
            "summary": "trunc"
        },
        "labels": {
            "sn": "_changeme_"
        },
        "condition": "B",
        "title": "SN000092 - Letztes Statusupdate",
        "data": [
            {
                "refId": "A",
                "queryType": "",
                "relativeTimeRange": {
                    "from": 600,
                    "to": 0
                },
                "datasourceUid": "_changeme_",
                "model": {
                    "expr": "(time() - (push_time_seconds{job=\"caterva_collector_SNxxx\"})) / 3600",
                    "hide": false,
                    "intervalMs": 1000,
                    "maxDataPoints": 43200,
                    "refId": "A"
                }
            },
            {
                "refId": "B",
                "queryType": "",
                "relativeTimeRange": {
                    "from": 0,
                    "to": 0
                },
                "datasourceUid": "-100",
                "model": {
                    "conditions": [
                        {
                            "evaluator": {
                                "params": [
                                    6
                                ],
                                "type": "gt"
                            },
                            "operator": {
                                "type": "and"
                            },
                            "query": {
                                "params": [
                                    "A"
                                ]
                            },
                            "reducer": {
                                "params": [],
                                "type": "last"
                            },
                            "type": "query"
                        }
                    ],
                    "datasource": {
                        "type": "__expr__",
                        "uid": "-100"
                    },
                    "hide": false,
                    "intervalMs": 1000,
                    "maxDataPoints": 43200,
                    "refId": "B",
                    "type": "classic_conditions"
                }
            }
        ],
        "ruleGroup": "_changeme_",
        "noDataState": "OK",
        "execErrState": "OK",
        "for": 1200,
        "orgId": 1
```

Look at the value for the field `for` which is set to 1200. The result should be 20mins. But when I check the result (`/api/ruler/grafana/api/v1/rules?subtype=cortex`) I get the value zero:
```
{
    "expr": "",
    "for": "0s",
    "labels": {
        "sn": "000092"
    },
    "annotations": {
        "summary": "Deine Caterva-Anlage hat seit über 6 Stunden keine Werte mehr an das Reporting übermittelt."
    },
    "grafana_alert": {
        "id": 12,
        "orgId": 1,
        "title": "SN000092 - Letztes Statusupdate",
        "condition": "B",
        "data": [
            {
                "refId": "A",
                "queryType": "",
                "relativeTimeRange": {
                    "from": 600,
                    "to": 0
                },
                "datasourceUid": "vXgO9xzVz",
                "model": {
                    "expr": "(time() - (push_time_seconds{job=\"caterva_collector_SN000092\"})) / 3600",
                    "hide": false,
                    "intervalMs": 1000,
                    "maxDataPoints": 43200,
                    "refId": "A"
                }
            },
            {
                "refId": "B",
                "queryType": "",
                "relativeTimeRange": {
                    "from": 0,
                    "to": 0
                },
                "datasourceUid": "-100",
                "model": {
                    "conditions": [
                        {
                            "evaluator": {
                                "params": [
                                    6
                                ],
                                "type": "gt"
                            },
                            "operator": {
                                "type": "and"
                            },
                            "query": {
                                "params": [
                                    "A"
                                ]
                            },
                            "reducer": {
                                "params": [],
                                "type": "last"
                            },
                            "type": "query"
                        }
                    ],
                    "datasource": {
                        "type": "__expr__",
                        "uid": "-100"
                    },
                    "hide": false,
                    "intervalMs": 1000,
                    "maxDataPoints": 43200,
                    "refId": "B",
                    "type": "classic_conditions"
                }
            }
        ],
        "updated": "2022-08-02T20:31:17Z",
        "intervalSeconds": 600,
        "version": 3,
        "uid": "oADDrbk4k",
        "namespace_uid": "5ODvrxzVz",
        "namespace_id": 5,
        "rule_group": "SN000092",
        "no_data_state": "OK",
        "exec_err_state": "OK",
        "provenance": "api"
    }
}
```

now the value is set to zero seconds. The value I send has no effect.

**What you expected to happen**:
I can controll the value of the field `for` over the api

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

Maybe this is related to https://github.com/grafana/grafana/pull/50992

**Environment**:
- Grafana version: 9.0.6 and 9.1.0-beta1
- Data source type & version: prometheus
- OS Grafana is installed on: docker
- User OS & Browser: Windows 11/ Edge
- Grafana plugins: none

**Comment by zuchka** [comment]
Thanks for creating this issue, @dfoxg 

Please check that you are using the newest stable API endpoints for provisioning Grafana alerts:

https://editor.swagger.io/?url=https://raw.githubusercontent.com/grafana/grafana/main/pkg/services/ngalert/api/tooling/api.json

also: is this cortex-specific? Thank you!

**Comment by dfoxg** [comment]
Hi @zuchka, thanks for your reply.

I'm using the latest API for provisioning. And no, this is not specific to cortex since I only use the prometheus data source.

**Comment by JohnnyQQQQ** [comment]
A fix for this is up and will be part of the 9.1 release, we will accept the `for` as duration. i.e. `60s`, `5m`, `1h` etc.

**Comment by dfoxg** [comment]
This change maybe introduced another issue: https://github.com/grafana/grafana/issues/55001
